ibis/expr/types/generic.py

```diff
@@ -2223,9 +2223,23 @@ class NullColumn(Column, NullValue):
 
 
 @public
-def null():
-    """Create a NULL/NA scalar."""
-    return ops.NULL.to_expr()
+def null(type: dt.DataType | str | None = None) -> Value:
+    """Create a NULL/NA scalar.
+
+    By default, the type will be NULLTYPE. This is castable and comparable to any type,
```

Comment:
```suggestion
    `NULL`s with an unspecified type are castable and comparable to values of any other type,
```

`NULLTYPE` isn't the name of anything in the Ibis codebase. I think it's probably going to be confusing that way.